Setup: make mavnative optional by default and update readme (#373)
ArduPilot/pymavlink: README.md

## Dependencies

> Pymavlink requires 4 dependencies :

Reviewer: ```suggestion
Pymavlink has several dependencies :
```

    - future : for python 2 and python 3 interoperability (http://python-future.org/)
    - lxml : for checking and parsing xml file (http://lxml.de/installation.html)
    - python-dev : for mavnative

>     - gcc : for mavnative

Reviewer: Will any C compiler work?

Author: no, both pip and setup.py will try to call gcc by default

Author: at least on linux

### For users

It is recommended to install pymavlink from PyPi with pip, that way dependencies should be auto install by pip.

```bash
sudo pip2 install -U pymavlink
```

The -U parameter allow to update pymavlink version if it is already installed.

#### Mavnative

> By default, pymavlink try to compile and install mavnative that is a C extension for mavlink python parser. Mavnative is quite old already and only support mavlink1.

Reviewer: ```suggestion
By default, pymavlink will try to compile and install mavnative which is a C extension for parsing mavlink. Mavnative only supports mavlink1.
```

> To skip mavnative installation and reduce dependencies like `gcc` and `python-dev`, you can pass `DISABLE_MAVNATIVE=True` environement variable to installation command :

Reviewer: ```suggestion
To skip mavnative installation and reduce dependencies like `gcc` and `python-dev`, you can pass `DISABLE_MAVNATIVE=True` environment variable to the installation command:
```

```bash
sudo DISABLE_MAVNATIVE=True pip2 install -U pymavlink
```

### For developers

On the pymavlink directory, you can use :

```bash
sudo MDEF=PATH_TO_message_definitions pip2 install . -v
```

The -v parameter will output the installation commands on the terminal.
The MDEF usage is require as pip install is done from /tmp directory, so it is necessary to use MDEF variable to 
point on the message_definitions directory.
Use pip should auto install dependencies and allow to keep them up-to-date with pip. 

> Or :

Reviewer: ```suggestion
Or:
```
